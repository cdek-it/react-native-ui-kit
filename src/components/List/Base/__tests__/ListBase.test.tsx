import { IconCheck, IconList, IconUser } from '@tabler/icons-react-native'
import { fireEvent, render } from '@testing-library/react-native'

import { ListBase, type ListBaseProps } from '../ListBase'

describe('ListBase tests', () => {
  const snapshotTestsConfig: Array<[string, ListBaseProps]> = [
    ['minimal config', { text: 'Menu Item' }],
    [
      'maximal config',
      {
        iconAlignment: 'top',
        text: 'Title',
        title: 'Subtitle',
        caption: 'Caption',
        LeftIcon: IconList,
        leftIconColor: 'red',
        RightIcon: IconUser,
        rightIconColor: 'green',
        extra: <IconCheck />,
        divider: 'content',
      },
    ],
    [
      'centered and full divider',
      {
        iconAlignment: 'center',
        text: 'Title',
        title: 'Subtitle',
        caption: 'Caption',
        LeftIcon: IconList,
        RightIcon: IconUser,
        extra: <IconCheck />,
        divider: 'full',
      },
    ],
    [
      'full disabled',
      {
        iconAlignment: 'center',
        text: 'Title',
        title: 'Subtitle',
        caption: 'Caption',
        LeftIcon: IconList,
        RightIcon: IconUser,
        extra: <IconCheck />,
        divider: 'full',
        disabled: true,
      },
    ],
  ]

  /**
   * Snapshot test with snapshotTestsConfig
   */
  test.each(snapshotTestsConfig)('ListBase %s', async (_, props) => {
    const rendered = render(<ListBase {...props} />)

    expect(rendered.toJSON()).toMatchSnapshot()
  })

  test('Calls onPress when pressed', async () => {
    const onPressMock = jest.fn()
    const { getByText } = render(
      <ListBase text='title' onPress={onPressMock} />
    )

    fireEvent.press(getByText('title'))

    expect(onPressMock).toHaveBeenCalledOnce()
  })
})
