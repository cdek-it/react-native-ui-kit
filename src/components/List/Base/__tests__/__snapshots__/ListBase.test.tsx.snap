// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBase tests ListBase centered and full divider 1`] = `
<View
  accessibilityLabel="Subtitle Title"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": "Caption",
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="ListBase"
>
  <View
    style={
      [
        undefined,
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 14,
          "paddingLeft": 14,
        },
        {
          "borderTopColor": "#dbdada",
          "borderTopWidth": 1,
        },
        false,
        false,
      ]
    }
  >
    <View
      style={
        {
          "alignSelf": undefined,
          "paddingVertical": 14,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={21}
        bbWidth={21}
        fill="none"
        focusable={false}
        height={21}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 21,
              "width": 21,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={21}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "type": 2,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
        >
          <RNSVGPath
            d="M9 6l11 0"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
          <RNSVGPath
            d="M9 12l11 0"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
          <RNSVGPath
            d="M9 18l11 0"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
          <RNSVGPath
            d="M5 6l0 .01"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
          <RNSVGPath
            d="M5 12l0 .01"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
          <RNSVGPath
            d="M5 18l0 .01"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      style={
        [
          {
            "flex": 1,
            "flexDirection": "row",
            "gap": 14,
            "justifyContent": "space-between",
            "paddingEnd": 14,
            "paddingVertical": 7,
          },
          {},
        ]
      }
    >
      <View
        style={
          {
            "flex": 1,
            "gap": 7,
            "paddingVertical": 7,
          }
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 12.25,
                "fontWeight": 700,
                "textTransform": "uppercase",
              },
              {
                "color": "#188700",
              },
              false,
              undefined,
            ]
          }
        >
          Subtitle
        </Text>
        <View
          style={
            {
              "gap": 3.5,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 15.75,
                  "includeFontPadding": false,
                  "verticalAlign": "middle",
                },
                {
                  "fontWeight": 400,
                },
                {
                  "color": "rgba(0, 0, 0, 0.8000)",
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Title
          </Text>
          <Text
            style={
              [
                {
                  "fontSize": 12.25,
                  "includeFontPadding": false,
                  "verticalAlign": "middle",
                },
                {
                  "color": "rgba(0, 0, 0, 0.6000)",
                },
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Caption
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 14,
            "paddingVertical": 7,
          }
        }
      >
        <View
          style={
            {
              "paddingVertical": 7,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={24}
            bbWidth={24}
            fill="none"
            focusable={false}
            height={24}
            meetOrSlice={0}
            minX={0}
            minY={0}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={2}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "type": 2,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth={2}
            >
              <RNSVGPath
                d="M5 12l5 5l10 -10"
                fill={null}
                propList={
                  [
                    "fill",
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  {
                    "type": 2,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={2}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={21}
          bbWidth={21}
          fill="none"
          focusable={false}
          height={21}
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 21,
                "width": 21,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width={21}
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          >
            <RNSVGPath
              d="M8 7a4 4 0 1 0 8 0a4 4 0 0 0 -8 0"
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "type": 2,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth={2}
            />
            <RNSVGPath
              d="M6 21v-2a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4v2"
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "type": 2,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth={2}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</View>
`;

exports[`ListBase tests ListBase full disabled 1`] = `
<View
  accessibilityLabel="Subtitle Title"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": "Caption",
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="ListBase"
>
  <View
    style={
      [
        undefined,
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 14,
          "paddingLeft": 14,
        },
        {
          "borderTopColor": "#dbdada",
          "borderTopWidth": 1,
        },
        {
          "opacity": 0.6,
        },
        false,
      ]
    }
  >
    <View
      style={
        {
          "alignSelf": undefined,
          "paddingVertical": 14,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={21}
        bbWidth={21}
        fill="none"
        focusable={false}
        height={21}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 21,
              "width": 21,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={21}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "type": 2,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
        >
          <RNSVGPath
            d="M9 6l11 0"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
          <RNSVGPath
            d="M9 12l11 0"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
          <RNSVGPath
            d="M9 18l11 0"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
          <RNSVGPath
            d="M5 6l0 .01"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
          <RNSVGPath
            d="M5 12l0 .01"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
          <RNSVGPath
            d="M5 18l0 .01"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      style={
        [
          {
            "flex": 1,
            "flexDirection": "row",
            "gap": 14,
            "justifyContent": "space-between",
            "paddingEnd": 14,
            "paddingVertical": 7,
          },
          {},
        ]
      }
    >
      <View
        style={
          {
            "flex": 1,
            "gap": 7,
            "paddingVertical": 7,
          }
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 12.25,
                "fontWeight": 700,
                "textTransform": "uppercase",
              },
              {
                "color": "#188700",
              },
              false,
              undefined,
            ]
          }
        >
          Subtitle
        </Text>
        <View
          style={
            {
              "gap": 3.5,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 15.75,
                  "includeFontPadding": false,
                  "verticalAlign": "middle",
                },
                {
                  "fontWeight": 400,
                },
                {
                  "color": "rgba(0, 0, 0, 0.8000)",
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Title
          </Text>
          <Text
            style={
              [
                {
                  "fontSize": 12.25,
                  "includeFontPadding": false,
                  "verticalAlign": "middle",
                },
                {
                  "color": "rgba(0, 0, 0, 0.6000)",
                },
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Caption
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 14,
            "paddingVertical": 7,
          }
        }
      >
        <View
          style={
            {
              "paddingVertical": 7,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={24}
            bbWidth={24}
            fill="none"
            focusable={false}
            height={24}
            meetOrSlice={0}
            minX={0}
            minY={0}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={2}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "type": 2,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth={2}
            >
              <RNSVGPath
                d="M5 12l5 5l10 -10"
                fill={null}
                propList={
                  [
                    "fill",
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  {
                    "type": 2,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={2}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={21}
          bbWidth={21}
          fill="none"
          focusable={false}
          height={21}
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 21,
                "width": 21,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width={21}
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "type": 2,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          >
            <RNSVGPath
              d="M8 7a4 4 0 1 0 8 0a4 4 0 0 0 -8 0"
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "type": 2,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth={2}
            />
            <RNSVGPath
              d="M6 21v-2a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4v2"
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "type": 2,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth={2}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</View>
`;

exports[`ListBase tests ListBase maximal config 1`] = `
<View
  accessibilityLabel="Subtitle Title"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": "Caption",
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="ListBase"
>
  <View
    style={
      [
        undefined,
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 14,
          "paddingLeft": 14,
        },
        {},
        false,
        false,
      ]
    }
  >
    <View
      style={
        {
          "alignSelf": "flex-start",
          "paddingVertical": 14,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={21}
        bbWidth={21}
        fill="none"
        focusable={false}
        height={21}
        meetOrSlice={0}
        minX={0}
        minY={0}
        stroke="red"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={2}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 21,
              "width": 21,
            },
          ]
        }
        vbHeight={24}
        vbWidth={24}
        width={21}
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
            ]
          }
          stroke={
            {
              "payload": 4294901760,
              "type": 0,
            }
          }
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeWidth={2}
        >
          <RNSVGPath
            d="M9 6l11 0"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4294901760,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
          <RNSVGPath
            d="M9 12l11 0"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4294901760,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
          <RNSVGPath
            d="M9 18l11 0"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4294901760,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
          <RNSVGPath
            d="M5 6l0 .01"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4294901760,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
          <RNSVGPath
            d="M5 12l0 .01"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4294901760,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
          <RNSVGPath
            d="M5 18l0 .01"
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4294901760,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      style={
        [
          {
            "flex": 1,
            "flexDirection": "row",
            "gap": 14,
            "justifyContent": "space-between",
            "paddingEnd": 14,
            "paddingVertical": 7,
          },
          {
            "borderTopColor": "#dbdada",
            "borderTopWidth": 1,
          },
        ]
      }
    >
      <View
        style={
          {
            "flex": 1,
            "gap": 7,
            "paddingVertical": 7,
          }
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 12.25,
                "fontWeight": 700,
                "textTransform": "uppercase",
              },
              {
                "color": "#188700",
              },
              false,
              undefined,
            ]
          }
        >
          Subtitle
        </Text>
        <View
          style={
            {
              "gap": 3.5,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 15.75,
                  "includeFontPadding": false,
                  "verticalAlign": "middle",
                },
                {
                  "fontWeight": 400,
                },
                {
                  "color": "rgba(0, 0, 0, 0.8000)",
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Title
          </Text>
          <Text
            style={
              [
                {
                  "fontSize": 12.25,
                  "includeFontPadding": false,
                  "verticalAlign": "middle",
                },
                {
                  "color": "rgba(0, 0, 0, 0.6000)",
                },
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Caption
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 14,
            "paddingVertical": 7,
          }
        }
      >
        <View
          style={
            {
              "paddingVertical": 7,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={24}
            bbWidth={24}
            fill="none"
            focusable={false}
            height={24}
            meetOrSlice={0}
            minX={0}
            minY={0}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={2}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "type": 2,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth={2}
            >
              <RNSVGPath
                d="M5 12l5 5l10 -10"
                fill={null}
                propList={
                  [
                    "fill",
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  {
                    "type": 2,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth={2}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={21}
          bbWidth={21}
          fill="none"
          focusable={false}
          height={21}
          meetOrSlice={0}
          minX={0}
          minY={0}
          stroke="green"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth={2}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 21,
                "width": 21,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width={21}
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup
            fill={null}
            propList={
              [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={
              {
                "payload": 4278222848,
                "type": 0,
              }
            }
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth={2}
          >
            <RNSVGPath
              d="M8 7a4 4 0 1 0 8 0a4 4 0 0 0 -8 0"
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "payload": 4278222848,
                  "type": 0,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth={2}
            />
            <RNSVGPath
              d="M6 21v-2a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4v2"
              fill={null}
              propList={
                [
                  "fill",
                  "stroke",
                  "strokeWidth",
                  "strokeLinecap",
                  "strokeLinejoin",
                ]
              }
              stroke={
                {
                  "payload": 4278222848,
                  "type": 0,
                }
              }
              strokeLinecap={1}
              strokeLinejoin={1}
              strokeWidth={2}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
</View>
`;

exports[`ListBase tests ListBase minimal config 1`] = `
<View
  accessibilityLabel=" Menu Item"
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="ListBase"
>
  <View
    style={
      [
        undefined,
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 14,
          "paddingLeft": 14,
        },
        {},
        false,
        false,
      ]
    }
  >
    <View
      style={
        [
          {
            "flex": 1,
            "flexDirection": "row",
            "gap": 14,
            "justifyContent": "space-between",
            "paddingEnd": 14,
            "paddingVertical": 7,
          },
          {},
        ]
      }
    >
      <View
        style={
          {
            "flex": 1,
            "gap": 7,
            "paddingVertical": 7,
          }
        }
      >
        <View
          style={
            {
              "gap": 3.5,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 15.75,
                  "includeFontPadding": false,
                  "verticalAlign": "middle",
                },
                {
                  "fontWeight": 400,
                },
                {
                  "color": "rgba(0, 0, 0, 0.8000)",
                },
                undefined,
                undefined,
                undefined,
                undefined,
              ]
            }
          >
            Menu Item
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 14,
            "paddingVertical": 7,
          }
        }
      />
    </View>
  </View>
</View>
`;
